### PR TITLE
Raspberry Pi support

### PR DESCRIPTION
**Pull request by mmstick**
Uses `if cfg!(aarch64) { "ports" } else { "us" }` to get the correct location for Ubuntu's source lists.

Also updates Cargo dependencies.

### CODE REVIEW

**Review by Absolucy — DISMISSED** [review]

File: daemon/src/release/repos.rs [hunk 1]
```diff
@@ -335,9 +335,10 @@ Types: deb deb-src
 URIs: http://us.archive.ubuntu.com/ubuntu/
 Suites: {0} {0}-security {0}-updates {0}-backports
 Components: main restricted universe multiverse
-X-Repolib-Default-Mirror: http://us.archive.ubuntu.com/ubuntu/
+X-Repolib-Default-Mirror: http://{1}.archive.ubuntu.com/ubuntu/
 "#,
-        release
+        release,
+        if cfg!(aarch64) { "ports" } else { "us" }
```
Comment (jackpot51) [inline daemon/src/release/repos.rs:341]: Shouldn't this be `cfg!(target_arch = "aarch64")`?

**Review by jackpot51 — DISMISSED** [review]

**Comment by linuxgnuru** [comment]
When trying to upgrade from 21.10 I got the following error:
```
system76@pop-os:~$ pop-upgrade release upgrade
checking if pop-upgrade requires an update
Current Release: 21.10
Upgrading to: 22.04
New version available: 9
Event: updating package lists
Release upgrade status: release upgrade aborted: failed to update package lists for the current release
```
Journalctl is attached, anything else I should send that could help?
[journal-pi.log](https://github.com/pop-os/upgrade/files/8959708/journal-pi.log)

**Comment by mmstick** [comment]
Upgrade successful on my Pi 400

**Comment by linuxgnuru** [comment]
i'll also test my 8GB pi 4 at home tonight

**Comment by linuxgnuru** [comment]
> Upgrade successful on my Pi 400

I'm curious, what resolution monitor are you using? I wonder if for some reason my work monitor might be too large and is slowing the pi down

**Comment by mmstick** [comment]
Using a VGA display with 1280x1024 resolution.

**Comment by linuxgnuru** [comment]
Hmm, I'm at 3840 x 1080 ... I'll try reducing it as well as try in a tty console and not gui; could also just be the factory internet

**Comment by linuxgnuru** [comment]
Ok, at home with my better internet and a 1440 monitor everything is upgrading perfectly.

**Review by linuxgnuru — APPROVED** [review]
upgraded in both gui and in multi-user.target and everything worked fine.

**Review by Absolucy — APPROVED** [review]

**Pull request merged**